ansible_anonymizer/anonymizer.py

```diff
@@ -268,15 +268,15 @@ def gen() -> Generator[str, None, None]:
 def hide_us_phone_numbers(block: str) -> str:
     flags = re.MULTILINE | re.DOTALL | re.IGNORECASE
 
-    def _rewrite(_: re.Match[str]) -> str:
-        return "(311) 555-2368"
+    def _rewrite(m: re.Match[str]) -> str:
+        return m.group("before") + "(311) 555-2368" + m.group("after")
 
     regexes = [
-        r"(?P<number>\b\d{10}\b)",
-        r"(?P<number>\b1\d{10}\b)",
-        r"(?P<number>\d{3}-\d{3}-\d{4}\b)",
-        r"(?P<number>\b\d{3} \d{3}-\d{4}\b)",
-        r"(?P<number>\(\d{3}\) \d{3}-\d{4}\b)",
+        r"(?P<before>[^\d\.])(?P<number>\d{10})(?P<after>[^\d\.])",
```

Comment:
I'm surprised we have five regexs just for a US phone number. Can this not be collapsed into one? Also, I'm not sure how we ended up only checking for US phone numbers. Shall we use a universal regex (or maybe a US regex AND a universal regex)?
I haven't fully digested each of these yet but they look like they're already acknowledging boundaries, looking for exactly X digits, etc, so it's hard to see where the exact problem lies that you're fixing. Examples/tests will help.

Reply:
@goneri probably we can go with only US number for now, and if we get issue around the other country phone number then we can implement this regex, where the catch would be your concern that `anything with spaces, dash and digit can look like a phone number.`